activerecord/lib/active_record/nested_attributes.rb

```diff
@@ -421,10 +421,15 @@ def _destroy
       # update_only is true, and a <tt>:_destroy</tt> key set to a truthy value,
       # then the existing record will be marked for destruction.
       def assign_nested_attributes_for_one_to_one_association(association_name, attributes)
-        options = nested_attributes_options[association_name]
         if attributes.respond_to?(:permitted?)
           attributes = attributes.to_h
         end
+
+        unless attributes.is_a?(Hash)
+          raise ArgumentError, "Hash expected for `#{association_name}` attributes, got #{attributes.class.name} (#{attributes.inspect})"
```

Comment:
Don't `attributes.inspect` risk leaking personal information into the error tracker / logs? Also is potentially very large.

I think just the class name is enough.

Reply:
Ya good point, I'll adjust.

Reply:
Actioned [here](https://github.com/rails/rails/compare/1cb4855aa1f76e0c1682bc071a286784c10accb9..1eab83a0a8d8c6b67ba83ab8e0abefa6ee8af4a8)